app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -47,7 +48,8 @@ def index
 
     def index_relation
       query = Product.readable.where(:organization_id => @organization.id)
-      query = query.where(:provider_id => @organization.anonymous_provider.id) if params[:custom]
+      query = query.custom if params[:custom]
```

Comment:
Do we still need to cast this to boolean. The false option for the params doesn't seem to filter correctly without it.

Reply:
So this is why I originally cast it as a boolean, but then that conflicts with the many katello API boolean params, see 'enabled' in that same call, which just assume 'true' on any value. 

So it seems the choice is to either conflict against how our API works now (cast as a boolean and use one param that accepts truthy and falsey values) or work with how our API works now (don't cast as a boolean and treat as truthy value). This is basically what we were getting at with [this discussion](https://github.com/Katello/katello/pull/7928#discussion_r248694580)

I can see the pros and cons of both ways and just went with two params to stay consistent to what we have currently. I'm open to a larger discussion around this area though, it brings up a good point.

Reply:
Makes sense..Works as intended treating any value as truthy as we do in other cases in our API..
ACK test ok! :+1:

Reply:
I still think we need to cast a boolean, the api was being handled incorrectly, 'false' should mean false, as should nil to be consistent with our api.  So to me, thats a bug in the api that shoudl be fixed as part of this.

Reply:
@jlsherrill, in that case, what is the point of having two params? What would `enabled=false` and `redhat=false` return?

Reply:
Think of each as a filter that is turned on or off,  'custom=true' only returns custom products, redhat=true (or redhat_only=true) only return redhat products.  Thats typically how boolean  filters behave

Reply:
@jlsherrill in the case of Products, aren't products either custom or redhat?

Reply:
they are, but I'm more talking about how our param filters typically behave, if set to true they filter the results, if not, they don't

Reply:
@jlsherrill That brings me back to my question, why have two boolean params for redhat and custom? There are either custom or redhat products, so it seems like you just need one boolean to filter that.

The more I think about it, I think this PR should go back to using only the custom param and cast it properly as a boolean with clear documentation. It seems we should start moving towards using boolean params properly by just modifying this existing one to work correctly.

Reply:
because booleans tend to not support 3 states as filter params, the three states being:

custom = true  (only return custom products)
custom = false  (only return  redhat products)
custom = nil (return both)

that can be confusing since all our other bool params consider nil to be false.  I'm not terribly opposed to this (as i said before), but i wanted to a acknowledge it and make sure it is well documented.

Reply:
You might could argue that 3 states makes the code a bit more complicated and less obvious:
```
if params[:foo]
   if cast_to_bool(params[:foo])
     #do foo
   else
     #do bar
   end
end
```

vs 

```
do_foo if cast_to_bool(params[:foo)
do_bar if cast_to_bool(params[:bar])
```

Reply:
@jlsherrill ok, I understand, false (in our current usage) means 'do nothing'.

Let me think about it some more, it probably is better to keep with the convention (2 parameters and cast to bool) for this PR though.

Thanks for explaining!

Reply:
oh i missed that you mentioned 'enabled=false', that does somewhat break convention based on the docs, but yeah it doesn't function properly with enabled=false, it just ignores that filter.  I think we need to clarify our documentation (and ideally naming on any new parameters)